cassandra/io/asyncorereactor.py

```diff
@@ -288,9 +291,12 @@ def _cleanup(self):
 
         log.debug("Event loop thread was joined")
 
-        # Ensure all connections are closed and  in-flight requests cancelled
+        # Ensure all connections are closed and in-flight requests cancelled
         for conn in tuple(_dispatcher_map.values()):
             conn.close()
+        # The event loop should be closed, so we call remaining asyncore.close
+        # callbacks from here
+        self._timers.service_timeouts()
```

Comment:
I realized while fixing this that asyncore.close callbacks where added to `AsyncoreLoop._timers` to be called later by the event loop thread but by this point it had already been closed.